### DISCUSSION TITLE
user specific sensors for TTGO T-Higrow with LoraShield

### DISCUSSION THREAD

**Discussion by hekopath** [cyberman54/ESP32-Paxcounter#732] [General]
Hello,

i try to customize paxcounter for the Lilygo/TTGO T-Higrow board.
This device has serveral sensors already on board.
And there are some good projects to make this information availabel with MQTT.
The [LoraWan example](https://github.com/Xinyuan-LilyGO/LilyGO-T-HiGrow-LoRaWLAN) is very simple and not power optimized.
This is why i tried to combine the paxcounting with more environmental monitoring.
I already managed to grab information from the ligthsensor (BH1750) and from a DHT11 (some newer version already have a bme280).
But i am stuck with the user sensor part.

Does someone have a simple example how to grab the sensor data and push it in the payload buffers?

If anybody is interested in an information exchange about the sensor handling?
As there are diffrent version in the market there is not the one and only way how to make the setup right.
Some discussion, for example about soil moisture sensor calibration, is welcome.

**Comment by cyberman54** [comment]
To add user sensor values to paxcounter:
- You only need to modify `sensor.cpp`, no need to touch other files
- In `sensor.cpp` assign your sensor's values and the number of bytes (= lenght) to the payload buffer:
    buf[0] = length;
    buf[1] = sensor_value1;
    buf[2] = sensor_value2;
    buf[3] = sensor_value3;
- If you need to read sensor_value_n from a sensor, add the appropiate code to `sensor_read(uint8_t sensor)` in `sensor.cpp`

**Comment by hansju-11** [comment]
and if you need to do a kind of setup or init your sensor once a startup you can use sensor_init() in sensor.cpp.

here is an example I did with scd30 (co2) sensor.

void sensor_init(void) {

  // this function is called during device startup
  // put your user sensor initialization routines here

   SCD30WIRE.begin();
  //This will cause readings to occur every two seconds
  if (! airSensor.begin(SCD30WIRE))
  {
    ESP_LOGE(TAG, "Check wiring von SCD30 Sensor");
    scd30_error = true;
  }
   
}

**Comment by hansju-11** [comment]
you can see a full quick example in pull request #780 .
an more complex example in pull request #787 .

**Comment by Dewald928** [comment]
I don't know if this is supposed to be obvious. But I have no idea which pins to use for the custom sensor on my TTGO T-Beam. 
I want to do a simple ADC read from a pin. Any ideas?

**Reply by cyberman54** [reply]
Check the board specs of your T-Beam and select one. This is not paxcounter related.

**Reply by Dewald928** [reply]
No problem. Would just like an example if anyone has done this with the paxounter

**Reply by cyberman54** [reply]
@Dewald928 
- You can follow the example in src/power.cpp how to read and calibrate analog value with ADC1 or ADC2 of ESP32
- Look at the code parts with #ifdef BAT_MEASURE_ADC
- Check T-Beam schematic for pins which are connected to ADC1 or ADC2 port, use ADC1 if possible

**Reply by Dewald928** [reply]
Thanks so much! I tried it, but my readings are incorrect. Trying to figure out why. I will report back here.
